subsys/nrf_rpc/rpc_utils/server/crash_gen.c

```diff
@@ -31,9 +34,11 @@ static void do_assert(void)
 
 static void do_hardfault(void)
 {
-	volatile uint32_t value;
-
-	value = 1 / 0;
+	__asm__ volatile (
```

Comment:
Why is this in asm instead of volatile variables division?

Reply:
volatile uint32_t value;
volatile uint32_t div = 0;
value = 1 / div;

The original problem may have come from the fact that "1/0" expression is constant.

Reply:
@MarekPorwisz Almost, it turns out you need to also put `1` into a volatile variable. Otherwise, the compiler is safe to assume that `div != 0` (as division by 0 is UB) and only checks whether `div` is 1 or not.